### PR TITLE
Ship matches for name stubs of `allowlist` for CUDA

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

File: easystacks/software.eessi.io/2023.06/rebuilds/2024.05.06-eb-4.9.1-CUDA-12.1.1-ship-full-runtime.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# 2024.05.06
+# Original matching of files we could ship was not done correctly. We were
+# matching the basename for files (e.g., libcudart.so from libcudart.so.12)
+# rather than the name stub (libcudart)
```
Comment (boegel) [inline easystacks/software.eessi.io/2023.06/rebuilds/2024.05.06-eb-4.9.1-CUDA-12.1.1-ship-full-runtime.yml:4]: Include link to this PR that shows what changed in hooks to fix this?

Comment (ocaisa) [reply easystacks/software.eessi.io/2023.06/rebuilds/2024.05.06-eb-4.9.1-CUDA-12.1.1-ship-full-runtime.yml:4]: Done

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Review by bedroge — APPROVED** [review]
I checked the log of the existing installation, and it contains:
```
== 2023-12-21 13:49:26,334 eb_hooks.py:455 DEBUG libcudart.so.12.1 is not found in allowlist, so replacing it with symlink: /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/CUDA/12.1.1/targets/x86_64-linux/lib/libcudart.so.12.1.105
```

For this new one, it has:
```
== 2024-05-06 17:07:27,084 eb_hooks.py:618 DEBUG libcudart is found in allowlist, so keeping it: /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/CUDA/12.1.1/targets/x86_64-linux/lib/libcudart.so.12.1.105
```

So, looks good to me.

**Comment by bedroge** [comment]
The tarballs have been ingested, and the symlinks look fine now:
```
$ ls -l /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/CUDA/12.1.1/lib/libcudart.so*
lrwxrwxrwx 1 cvmfs cvmfs     15 mei  6 19:05 /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/CUDA/12.1.1/lib/libcudart.so -> libcudart.so.12
lrwxrwxrwx 1 cvmfs cvmfs     21 mei  6 19:06 /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/CUDA/12.1.1/lib/libcudart.so.12 -> libcudart.so.12.1.105
-r-xr-xr-x 1 cvmfs cvmfs 679264 mei  6 19:06 /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/CUDA/12.1.1/lib/libcudart.so.12.1.105
```

**Pull request merged**